sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -56,16 +77,69 @@ def test_silhouette():
             assert_almost_equal(score_euclidean, score_dense_with_sampling)
 
 
+def test_silhouette_invalid_block_size():
+    X = [[0], [0], [1]]
+    y = [1, 1, 2]
+    assert_raise_message(ValueError, 'block_size should be at least n_samples '
+                         '* 8 bytes = 1 MiB, got 0',
+                         silhouette_score, X, y, block_size=0)
```

Comment:
Why do not select automatically `block_size` as the `min(64MB, np.ceil(n_samples * BYTES_PER_FLOAT * 2 ** -20)` and don't let the user choose a specific value ?